### DISCUSSION TITLE
[Feature] Duplicate Rules

### DISCUSSION THREAD

**Discussion by johnny2678** [immich-app/immich#10665] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Great job with the duplicate feature.

I have 75,000 dupes!  So while I click away, I have had some time to think about how the "recommendations" might be more finely tuned.

ex.  If photo resolution is the same and a HEIC and JPG show as dupes, immich usually recommends the JPG as the file size as bigger.  However, I prefer the HEIC because of the compression and the live photo

another one - if an item exists in an external library and in the immich library and the resolutions are the same, I prefer the external library version.  

I'm sure others have different opinions, but that's why it would be nice to be able to define our own rules.

Thanks for the project!

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by marciopamplona** [comment]
Another very useful feature is enable/disable video dedupe. I have a lot of false positives due to the AI model.
It could be very useful to have a different configuration for detecting videos with a more strict rule (different maximum distance).

**Comment by radh21301** [comment]
Here are some more points to think about. It would be great if we could set a list of rules and prioritize them. And each user can have a different priority - basically, an option to reorder them as needed.

The following are some rules I could think of:

1. GPS Available (Yes/No)
2. Exif Tag Available (Yes/No). 
    - *DateTimeOriginal:* 
         - I have this criterion because there could be pictures with the metadata stripped, say a WhatsApp image. These images that don't have the metadata will most certainly show up in an odd location in the timeline. I always prefer images with the right metadata.
    - *CreateDate* - This is for videos
    - *Starred*
    - The user could choose any tag from a certain pool of most commonly used exif tags
4. Image/Video size (Small/Large)
5. Prefer external library OR immich internal library
6. Prefer Immich Favorited Asset

The list can be refined based on what people think.

**Reply by premiumcola** [reply]
This would be awesome :)! thanks for your detailled thoughts on this topic

**Reply by Rundervinkje** [reply]
I'd like to add-on to your list with:

1. Duplicates (same name). Like D001_img2.jpg and D001_img2.jpg in a different folder.
2. Duplicates (partly same name). Meaning, it could even account for partial matches, like recognizing D001_img2.jpg and attachment_D001_img2.jpg as the same image despite slight differences in the name.
3. Duplicates per Maximum Detection Distance (MDD). Meaning, run the duplicate search job with different MDD's and sift through them at a later date.

**Reply by cookie-monster1649** [reply]
I'd also love to add in
- Is in album (yes/no)

**Comment by tairasu** [comment]
this is fairly easy to implement since the suggestDuplicateByFileSize object is just sorting each iteration by file size and selects the last object (biggest file). instead you could first check if there is one or more HEIC file and then ignore JPGs.

obviously this should be optional. and it would need a UI for user preferences.

**Comment by Ccheese4** [comment]
Also been thinking about this as I go through my 13,000 duplicates. Would at least reduce my clicks so I don't have to switch the heic and the jpg auto selection.

**Comment by Ccheese4** [comment]
also the ability to skip a duplicate pairing would be nice and come back to it later.

**Reply by maicss** [reply]
I really need this. Do you think we should begin a new conversation about this?

**Reply by maicss** [reply]
I created a new discussion about this. https://github.com/immich-app/immich/discussions/17566

**Reply by brucethomax** [reply]
How would this "skip" be different than just clicking the "next" button? Would that just be button placement? I am not a fan when buttons jump around on the screen which happens to the lower button group since the container snaps to the size of the assets.

**Comment by chrissi5120** [comment]
I effed up big time and uploaded from my wifes "recent" folder which seems to be populated by google photos?

We now got like 8k photos and video as duplicates with/without GPS. 

being able to select by GPS would save me tears.

**Reply by githubsean** [reply]
What I would like is if some information from the deleted duplicate is copied over to the kept duplicate. A good example is folders.

Say there are two photos. One is low resolution (from Google Takeout), one is high resolution (original manually copied off phone). The low resolution version is in an album, the high resolution one is not in any album. The duplicate checker will correctly suggest to keep the high resolution version, but the effect is to remove this photo from the album the low resolution version was in.

Similar issues might happen with EXIF dates or GPS coordinates - the photo being discarded might have this detail, but the kept photo might not (though given the example above, the "original" will likely have all this). My main issue is with keeping album information.

**Comment by Fireflaker** [comment]
Same request here! 
Deduplication - keep xxx.HEIC while delete xxx.HEIC.jpg

**Comment by njspix** [comment]
Wanted to bump this - file size is a decent criteria but might end up selecting for poor compression algorithms (JPEG instead of HEIC, for example) rather than *actually more data*. For a preliminary go, how about sorting by (decompressed) pixel count rather than size? I believe it's possible to get the 'real' pixel dimensions of HEIF/HEIC and JPEG images without decoding them.

**Comment by JasonMAnderson** [comment]
A dilemma I run into is camera backup from my phone.  I want my photos from my phone immediately available, so I turn-on camera backup.  But I also manually copy my camera photos to my external library every month because my external library is my "single source of truth" as it's accessible outside of Immich.  I need a rule to always delete the duplicate in the camera backup if a dupe exists in external library.

**Reply by premiumcola** [reply]
Yeaaah thanks for the comment on this aspect:
I would also like to set my "external libary source" as first priority - and keep all fotos once in there but delete the duplicates on all other sources.

**Reply by jcisio** [reply]
You can also move your uploaded photos (you should use a time based template to make it easy) to the external library, instead of moving your camera photos?

**Comment by rwlove** [comment]
Prefer to keep photos already in an album.

**Comment by nicolas-albert** [comment]
Hi. I'm a Google Photo use from years. I have my home Immich server and I push to it my 450 Go of media from Google Takeout.
Each photos edited for cropping or color are duplicated (+10000).
Also currently saved from my phone (+4000 to save) already here with the Takeout import and bigger.
But I prefer smaller (cropped and better compression) and already in album.
I hope this feature will be selected 👍

**Reply by jcisio** [reply]
For external library we should be able to mark some directories as "preferred source". For uploaded photos: we should be able to mark some albums as "preferred source". If this is implemented then it would solve your problem.

**Comment by andredatele** [comment]
Another one bumping this feature. I'm using immich-go to import from Google Photos, but the dupe detection is not perfect, leaving most of the work to be done in Immich itself. Would be great to have the ability to prefer assets already in albums.

**Comment by nicolas-albert** [comment]
[immichDuplicate.html.zip](https://github.com/user-attachments/files/19463441/immichDuplicate.html.zip)
I created a bookmarklet to help me choose simple cases.
It searches for French words and needs to be adapted.
It auto-validates every second and stops on ambiguous cases (until pressed again).
It's better to understand it before running it.

**Reply by killforby** [reply]
Bonjour comment exécuter le bookmarket ?

**Reply by nicolas-albert** [reply]
Bonjour,
De souvenir, il faut l'ajouter dans la barre de favoris. Une fois sur la page de recherche de duplication sur Immich, on clique dessus et ça va choisir une image suivant des critères (dont je ne me souviens plus) et virer les autres, puis passer à la suivante jusqu'à tomber sur un cas que le script ne gère pas.

**Reply by killforby** [reply]
D'accord je vais voir ça, vraiment dommage qu'il intègre pas la suppression par étiquette dans immich directement

**Reply by killforby** [reply]
Et merci pour ta réponse

**Comment by shanep2300** [comment]
I want the option to be able to 'skip' and 'ignore'. Currently I have multiple albums that are of different resolutions but same file name. IE: WebRes, FullRes, Raw. I don't want any of those removed.

Maybe be able to see which albums the dupes are in and then uncheck it from looking in there anymore?

**Reply by brucethomax** [reply]
Wouldn't the stacking feature work here? If images are stacked, the duplication detection moves on. Also, if you "keep" rather than "skip" with next, it should be marked as resolve and not come up again. Or am I missing something?

**Comment by qkrrmsdud1** [comment]
An "undo" button and the ability to not classify assets as dupes if they have different filenames (e.g. IMG_1234 and IMG_1235) would be nice as well.

**Reply by GiorsAltF4** [reply]
yeah this is literally all I'd need, Ctrl-Z to undo and a way to filter the duplicates type found (so for example if they have different names they don't appear in the filtered view)!

I wonder if there's any way to bump this suggestion

**Reply by brucethomax** [reply]
Ctrl-Z stacked for how many photos? I'd prefer a redesign were there was a list of duplicates and only until you do a bulk operation does it actually do trash them. Then an "undo" feature that would take a look at all of those to restore for that session. Those are two different components though, increasing complexity.

For me personally, I'd settle for a list/bulk redesign and the ability to sort the trash by a "trashed on date" (currently the asset date is used) to restore selected assets. What do ya'll think?

**Comment by punqbund** [comment]
As mentioned initially, regarding de-duplication: Select one or more preferred libraries for resolving duplicates. Images from these libraries will be selected to keep (or will be automatically kept in case of resolving all duplicates), and images from all other libraries will be deleted unless you define additional criteria.

**Reply by chrissi5120** [reply]
seconded. The focus should be exactly here. 

The simplest situations like checking duplicates between multiple libraries OR albums should be implemented first.
Next priority could be quick-wins like whatever above is also straightforward to implement.

Another big focus should be in UI/UX, the more you can fix here, the better.

**Reply by brucethomax** [reply]
I'd second the "keep the one in the album if the other assets are not" feature! 👍 Easy enough!

Though I don't know about the "multiple libraries" check. Can you elaborate on that a bit more?

**Comment by hday0** [comment]
As someone who is having to go through 8000+ duplicates found by the utility manually, this would seriously be a god send. I want to add a rule to prioritize auto selection of a date timestamp instead of a generic "img_2492.jpg"

**Comment by plsander** [comment]
I would find ability to review dups associated with images in a particular album useful.

**Comment by mmomjian** [comment]
#10537

**Comment by sebastiencastermans** [comment]
Hey everyone!
I made a [Python script](https://github.com/sebastiencastermans/immich_duplicates_python_script) to help clean up duplicate photos and videos on Immich servers. It smartly prioritizes Apple’s HEIC files and offers a dry-run mode so you can preview before deleting anything.
If you’re looking to tidy your Immich library safely and efficiently, feel free to check it out! Happy to get feedback or ideas for improvements.
I hope it will be usefull to you :)

**Comment by joekueng** [comment]


**Reply by bo0tzz** [reply]
@joekueng mangling things through an LLM isn't very useful, it's made up a bunch of things that aren't implemented so the basis that your proposed PR would build on doesn't exist.

**Comment by joekueng** [comment]
Hello everyone, I have implemented this to prioritize photos from the external library.

Obviously, I still need to test it thoroughly, but does it make sense as a first approach? Also in terms of the UI.

What do you think of this approach?

<img width="1281" height="674" alt="Screenshot 2025-09-01 at 21 53 36" src="https://github.com/user-attachments/assets/44181f59-38c5-41be-8b24-44a5a495c266" />
<img width="1281" height="674" alt="Screenshot 2025-09-01 at 21 53 44" src="https://github.com/user-attachments/assets/4f4bf464-fc45-44c1-8295-0af9a721276e" />

**Reply by JasonMAnderson** [reply]
Dear god - PLEASE implement this.  It would solve most of my 5K+ duplicates

**Reply by jjmonester** [reply]
Tbh, I have the opposite problem. I want to prioritise the internal library. Due to re-uploads through the app, and adding to albums, internal photos are the ones I want to maintain. 

It'd solve the problem for me if there was a toggle to prefer internal or external libraries. Having this toggle between the two would be a godsend.

**Reply by joekueng** [reply]
@jjmonester I can implement it, thanks for the idea. Do you have any ideas on how to do it for the UI?

**Reply by jjmonester** [reply]
I haven't thought deeply about the intention of this whole tool, but when I think of the users of Immich, generally everyone has slightly varied setups and are technical power users (at least those using the deduplication tool). I think we'd need to account for this. 

I'd assume then this is probably one of many filters that users would want to set to change how the algorithm works for default deduping. 

For the simplicity of this implementation, I'd create a settings dropdown with defaults set to how Immich is now as 'Default'. This setting would then effectively override the logic set by the OG devs, so I'd want to know explicitly if my settings are changed when using the tool. 

So some rough requirements
- A Settings dropdown button for the dedupe utility
- A dropdown for this setting. Could be something like 'Prefer assets in: Default, Internal library, External Library'
- Create some kind of visible prompt, a colour change perhaps on the settings dropdown button, if any settings deviate from the Immich defaults. This is to signal to users that a change from defaults is currently in effect.  
- Create a confirmation prompt (modal) on pressing 'Deduplicate All' confirming that the settings have been altered from the defaults. 

TL;dr
- Maintain clear defaults for standard Immich behaviour
- Allow for future customisation and detailed utility changes
- Warn users for changes in place away from default. 

I'm not a designer so can't help much with that sorry! I'm a product weirdo myself 😊

**Reply by jjmonester** [reply]
To just add on, there's so much discussion above about other variations and extra metadata to add to help sort. I think we start simple as above and go from there. 

If a dedupe tool is effectively a list of ordered rules like @radh21301 , that could be a good future state to move towards. A rule builder of sorts where one sets the logic of the underlying system so you can make it work for your specific use case. But we'd need the metadata exposed and usable first. 

Personally, I mainly care about maintaining any albums / favourites and additional metadata added to images. If there was a way to ensure that info is maintained regardless of asset, that would be great too.

**Reply by joekueng** [reply]
Looking at the other utilities, I don't think they are mature tools. There should be a section in the settings to change the parameters. 

However, I believe that covering the three cases:

- prefer external
- default
- prefer internal

is necessary in order to use the deduplication tool.

In the future, it will certainly be necessary to change the logic and implement new settings and functions as discussed in #10537.

**Reply by SailLively** [reply]
Who cares if it is internal or external, I just want the highest quality photo with the correct date and metadata, I would like to prioritize the oldest photo and have the metadata merge functions expanded so we can choose the correct date, best photo, and the correct name all independently from the slew of files

**Comment by joekueng** [comment]
I changed the way here you select the preferred libbrary, now you can choose between internal external or leave default.

<img width="1070" height="572" alt="Screenshot 2025-09-02 at 11 39 19" src="https://github.com/user-attachments/assets/b4889d6d-e622-45ae-b4ac-8e577e051b76" />
<img width="1070" height="572" alt="Screenshot 2025-09-02 at 11 39 16" src="https://github.com/user-attachments/assets/4b71fc2e-2c6e-4e0b-84bc-d9caee61b72a" />

The logic prior to preference has not been changed.

What do you think of this implementation? @jjmonester @bo0tzz

**Reply by jjmonester** [reply]
Sure, I think that's a good first implementation! Really nice work to simply enable the functionality 😄 

If I were the official dev team for this, I'd probably want it to be future proofed with a space for more alterations of the dedupe logic in its implementation, ie, a dropdown. I'm imagining a modal, similar to when you click on the user's profile picture top right of the interface. Then that would mean the modal can expand into the future with different options instead of cluttering the UI. 

 Is that still possible?

**Reply by zurkiyeh** [reply]
@joekueng when can we expect this to be released please?

**Reply by cosmickylenny** [reply]
Looking forward to this. I want to prioritise internal files. I have about 2000 duplicates

**Reply by joekueng** [reply]
Hi everyone, I'm glad you're looking forward to the new feature. I've opened a pull request and am waiting the review.
#21595 

@cosmickylenny @zurkiyeh

**Reply by zurkiyeh** [reply]
Looks great thanks a lot!

**Reply by soylentseth** [reply]
that is perfect for my needs.  thank you sir

**Reply by The-Croz** [reply]
Could this be supplemented with an option to prefer HEIC images over JPEG? Have been looking for this option but have not found it anywhere.

**Reply by joekueng** [reply]
@The-Croz
I'm waiting for the pull request to be approved, and then I'll definitely try implementing the HEIC preference.

**Reply by The-Croz** [reply]
@joekueng that would be fantastic. Appreciate your work on this. Hopefully they merge your PR soon!

**Comment by agentgodzilla** [comment]
The critical duplication "rule" for me that is missing is the ability to exclude certain albums (id take an option to exclude all of them) from comparison.  For example I have my wedding photos in my immich which due to the nature of them I have ~1000 items that need deduplicating, but 99% of it is photos I definitely want to keep.

**Comment by cosmickylenny** [comment]
any news on when this will be available?

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by dhlavaty** [comment]
Duplicate Tool could not only delete photos, but also merge META information.

I have a lot of duplicates in my library, and it often happens that a photo I want to keep because of its image, does not have all the EXIF data that the photo I am deleting has. This would add the EXIF information from the deleted photo to the one I want to keep.

It would be great if we could consider the Duplicate tool as a merge tool — something like Merge Contacts in your Contacts app.

**Reply by punqbund** [reply]
In my opinion, if there is a need to store and combine metadata from a deleted duplicate, this should only be done in the database. The link between the metadata of the deleted duplicate and the file you are keeping should also only exist in the database. Writing metadata to an image would be file editing. In my opinion, editing files goes beyond the scope of a photo library, at least in the short term. This is my experience based on the development of DAM systems. If you change the original, the question of undoing the changes naturally arises, which in turn means version control, and so on. Such things can be planned for the long term, but for now, you can think about storing and merging metadata from a deleted duplicate with a link to the preserved file at the database level.

**Reply by punqbund** [reply]
And keep in mind, that removing duplicates doesn't always mean deleting them, but rather removing the database entries.

**Reply by dhlavaty** [reply]
Agree of course. No image file manipulation in any way.

**Reply by brucethomax** [reply]
Then what happens when a "rescan of metadata" is triggered? Wouldn't it be overridden? I can jump into the code if no one knows off the top of their head.

**Comment by antonio-ivanovski** [comment]
Similar to the proposal from @dhlavaty, often times there is duplicate with lower quality that has ended up in an album but the better one is not in album. This is often the case when someone has performed Google Photos upload and also managed to get the original photos uploaded too.

Proposing option when a deleted photo was in a album, there to be option to be replaced with some of the photos that were kept.

<img width="708" height="267" alt="image" src="https://github.com/user-attachments/assets/dd40ff92-2efa-43c9-9763-4110887a8971" />

**Comment by Tim-Dooley** [comment]
I hope this feature gets implemented soon with some options.
I have many duplicate files where the lower quality one is in the album.
Personally I would like the following order of operations for duplicates
1. highest resolution first
2. heif/heic over JPG
3. Smaller files over larger files
As an add on, merge album links.

This would mean if the higher resolution file is jpg then discard the heic
If resolution is equal between a jpg and heic discard the jpg.
If resolution is equal & file extensions match, keep the smaller.

**Comment by MacPaKKo** [comment]
I am removing duplicates.
I'd like to have options to sort proposed duplicates by date, so photos will follow a logical timeline, or by size (movies for sure come first), so I'll quickly clean space on my disk (following delete rules).
Thanks

**Comment by emplexity** [comment]
Thanks for the duplicates handling in immich (version 2.5.6).
I understand that:
. the "deduplicate all" trash all duplicates from immich and file system, which can be quite destructive if pic's have been assigned to albums before.
. the "keep all" option is skipping the duplication handling altogether
Could an additional option "Stack all" be added to process softly all duplicates in one go so that the album assignments of pic's are not affected?
Is there some doc's that a user can access to understand the consequence of each option?
Thanks

**Comment by brucethomax** [comment]
My request was **closed** before a conversation could be even had: https://github.com/immich-app/immich/discussions/26672

This feature proposed here is extremely involved! I am suggesting a simple default update (while _this_ feature is mulled over and implemented) that will enhance the typical manual migration strategy. I have several different archives from iCloud, Google and Adobe that have each _potentially_ tweaked the file and when exported/imported. When importing into other catalog tools (such as Lightroom) they will automatically increment the filename since the duplicate detection (in Lightroom) is weak. These programs typically add rather than remove char from the filename. 

I am proposing a **_simple_** 3-5 line adjustment at the end of the duplicate checked (`duplicate-utils.suggestDuplicate`) after the file size and exif checks. If all things are the same, take a look at the originalFileName and keep the shorter one. Reproducible, clear, and what I believe should be the default.

**_What do you all think?_** Should I submit a PR to articulate? I'd like to do this smaller change to get familiar with the development process then I'd like to take a stab at making the duplicate detection process better in Immich. I want to contribute!

I see this being an easy win by default, currently it is random! Cases where one has manually updated assets names to be something else, then you will need the full feature of duplication rules.

<img width="1048" height="635" alt="Screenshot 2026-03-03 at 22 39 19" src="https://github.com/user-attachments/assets/a969b681-e60b-4c2e-9698-5234720da5e2" />

The **question** I have for everyone is, all things considered would you prefer to keep shorter filenames (vote with 👍) or longer ones (vote with 👎)? Please prove examples for either case, though especially the 👎 case.

**Reply by mmomjian** [reply]
It is unlikely that we would accept a PR for such a specific use case.

**Reply by brucethomax** [reply]
Sure, I get that unless enough people want it since there is currently no default and the selection is hidden in this specific example. 

In the meantime, I pulled everything down and started working on the larger feature of rules and such. I'll play around with it locally for a few days since I don't want to waste my days deduping my photos by hand.

This is such a beautiful designed project, I can't believe how easy it was to get started! 😍

I'll make a post here when I get something solid to see what input people might have. 

I don't see any updates or anyone working on this or did I miss something?

**Comment by brucethomax** [comment]
### Proposed Design 🚀
I have merged a lot of different cloud exports (Google, iCloud, manual exports before there were any cloud options, etc) that have been in varying states of decay. 

I have taken a stab locally, though I know that this will require a bit of feedback and input. I am not sure if what I have here will ever land or come out the other side. This is what I have started with.

### Duplicate Groups List
Right now there is no way to just look over the the duplicate groups to get a since of where to start. I have added a list with some basic information and the ability to sort it. If there is some value in the list alone, it can even be added with simple additions leaving the duplication logic untouched. What do you think?

<img width="1186" height="766" alt="Screenshot 2026-03-09 at 18 14 38" src="https://github.com/user-attachments/assets/e87cad0a-3300-4bb1-8ac1-f0680840a112" />


Here are the current sorting strategies of the duplicates.

<img width="309" height="432" alt="Screenshot 2026-03-09 at 18 09 56" src="https://github.com/user-attachments/assets/5e608fd4-d225-42af-859d-58d3a91968e0" />


I like the sorting by count and size to eliminate egregious offenders first!

<img width="314" height="345" alt="Screenshot 2026-03-09 at 18 08 34" src="https://github.com/user-attachments/assets/12e332ab-450c-4c5a-8d16-9a11592af203" /><img width="314" height="308" alt="Screenshot 2026-03-09 at 18 09 40" src="https://github.com/user-attachments/assets/5baf9f84-f641-4a2e-953e-07ea7ca04354" />

### Duplication Rules
Currently, all logic I have is all on the frontend and the concept I have here is the least polished. I'll be working on this over the next weeks locally on my various collections. I'll be diving into this thread and incorporating a much of what was said here of the different rules. 

The ultra basic concept here is that these rules can be prioritized with a simply point system (which I am still working on.. ideas welcome).

Stay tuned for a complete list (perhaps an MD in a branch) of rules and options.

<img width="652" height="815" alt="Screenshot 2026-03-09 at 18 50 52" src="https://github.com/user-attachments/assets/0c50585c-9b2f-4695-ad3a-39496be6c867" />

This is something that I added that helped me out during my duplication search. A **reviewed state** that can then isolate your duplication deletions in a batch (which could be offloaded to the backend in the future) with this PR (https://github.com/immich-app/immich/pull/25316).

<img width="311" height="367" alt="Screenshot 2026-03-09 at 18 17 00" src="https://github.com/user-attachments/assets/8fca4832-8db7-4d79-a528-e58a23feb8a3" /><img width="442" height="367" alt="Screenshot 2026-03-09 at 18 40 17" src="https://github.com/user-attachments/assets/6af1f069-fe0d-4d68-b57b-1e033318fa45" />


### Updates to the Compare Details
The notable updated here is that I added a couple metadata fields for the camera. There is another PR that will clean up fields that are the same on all duplicates (https://github.com/immich-app/immich/pull/26328).

Note the ✅ and ❌ that indicate which rules were passed.

<img width="1015" height="331" alt="Screenshot 2026-03-09 at 19 51 11" src="https://github.com/user-attachments/assets/7ce14ccb-33c4-44c5-8e99-0437d27c9681" />


### Other Considertaions
Currently only a single "keeper" is selected, though I am open to ideas on how to selected multiple within a category. For instance, if there is a original and an "edited" version though there might not be a clean way to do this reliably.

Feedback would be greatly appreciated. This is a rough concept at the moment yet is already a large change though isolated. I'd like to not add complexity, just refine the concepts here or even identify low hanging fruit 🍎 that can be plucked. Thanks!

**Reply by multi-suggester** [reply]
2 things 1 is i think adding thumbnails to the slider would be nice  (i think itll either has to be very small or youll have to change the shape of slider ) (might wanna load images gradually , i have 2.7k duplicate).

2 is i think by using some sort of toggle we can halve the amount of options and decrease the visual clutter/complexity? , having a line for each option is a but not good looking imo
maybe something like in photo?
<img width="169" height="39" alt="image" src="https://github.com/user-attachments/assets/2b905957-c3d8-4070-9802-6fd3bb36f6eb" />

id like to comment on duplication rules but i think ill need to comprehend before that as its not something i normally use

**Reply by SimonRiemertzon** [reply]
I think this looks awesome! Especially the priorities of how the duplicates get selected. 

I have recently pulled all my photos from Icloud Photos and Google Photos, because --- many many reasons ---. As I have done that I have gone for the "better safe than sorry"-strategy which have made me have a lot of duplicates and I think immich current capabilities to sort them are subpar. It will take me ages to sort through. I'm sure that this would speed things up significantly and be a huge help for a lot of users. I'm now having to decide if I should go try out some version of jdupes, fdupes, czkawka, implement something myself with exif-tools or waiting patiently for your PR while cheering you on! 😍

Some feedback:

1. **Birds-eye-view** - I feel like its hard to get an overview of which duplicates there are and sort through them quicker. I'm not sure about the usefulness of this, but using current immich solution, looking at one instance of duplicates per page refresh... yeesh. 

2. **Keep RAW originals rather than compressed** - Since I value RAW files higher, (I could always re-compress and export them), I often want to delete jpg versions of photos. A big caveat though is that if I have edited the photo in lightroom I would most definitely want to keep both the rew and exported version. This is where "created date" would come in I guess. It might be enough just to add a file type to your priority list.

**Comment by DMC1285** [comment]
I have this same request similar to the original post due to multiple backups and JPEG / HEIC of the same photo.  While there are many additional filters here, at the core of it all I just want to be able to select the file types that I want to keep of the highest quality and remove the rest (HEIC and JPEG of the same picture).

**Comment by BPier** [comment]
<img width="1081" height="707" alt="image" src="https://github.com/user-attachments/assets/bf40cf99-86ac-44f2-aa03-9af7a598a66f" />

About the criterias for selecting the default duplicate to keep, I would add the date as well.

In my case, I have plenty of pictures that has been copied in multiple folder. The pictures are strictly identicals, only the file date is changing. I would like to keep only **the oldest picture**, not the most recent one.

**Reply by hh7418695** [reply]
same situation as mine

**Comment by vworldat** [comment]
+1 for allowing to skip/hide certain duplicates based on custom filters. My immich account currently displays ~10k duplicates. 95% of those are RAW+JPEG file combos or multi-resolution exports from my external library **I want to keep on purpose**. I don't see the point to wade through >9000 duplicate checks that I KNOW to be there and valid while actually caring about a handful of accidental duplicates that I'd actually remove.

Filters that would make my life easier:
- Completely ignore specific storage path and/or file name patterns
- Ignore duplicates that don't include at least 1 item in my main library. If they are both on the external lib I don't care. If I sent a photo from the external lib via WhatsApp, I want to be able to discover that specific duplicate so I can remove the WhatsApp copy.
- Sort / filter by date

These kinds of filters would be awesome and, importantly, not touch the actual duplicate checks at all. I don't mind my server having to wade through all the RAW images and tag them as duplicates, I just don't want to see them in the duplicates tool by default or at least be able to hide them. Sometimes I'd even like to be able to see all duplicates of a specific RAW file so I can check for actual exports, but on a by-picture level.

At the moment the inability to suppress certain duplicates from showing up makes it unusable for me 😞. I guess this also links to #27458

**Reply by antoniovj1** [reply]
Same here, I have like 90k duplicates, almost all of them JPEG + RAF (Fujifilm RAW) but these pairs should be ignored

**Comment by Dieterhomeass** [comment]
I would like to have a toggle switch next to the trash button, because many pre-selections are the wrong way for me.

**Comment by Gasdobun** [comment]
I don't know if this could be a feature request, but has it been considered to create "custom filters" for which file to be kept? 
I, for instance, would love that the kept file has the older timestamp, since due to a lot of... things, the "recent" copy is the one who became metadata-rich -and with a larger filesize, thus being the default kept one-, but to keep the dates the kept image should be the older, that would then get the metadata synchronization, but doing that manually when the duplicates might be a very big list makes it tiring.

Should I phrase it better for a feature request? Should it be talked here? 

thanks in advance, any corrections on my words are welconme since English isn't my first language, lol
